include/enrico/openmc_driver.h

```diff
@@ -95,10 +97,10 @@ class OpenmcDriver : public NeutronicsDriver {
 
 private:
   // Data members
-  openmc::Tally* tally_;               //!< Fission energy deposition tally
-  openmc::CellInstanceFilter* filter_; //!< Cell instance filter
-  std::vector<CellInstance> cells_;    //!< Array of cell instances
-  int n_fissionable_cells_;            //!< Number of fissionable cells in model
+  openmc::Tally* tally_;                     //!< Fission energy deposition tally
+  openmc::CellInstanceFilter* filter_;       //!< Cell instance filter
+  std::map<CellHandle, CellInstance> cells_; //!< Array of cell instances
```

Comment:
I implemented `OpenmcDriver::cells_` as a `std::map` so I didn't have to assume the CellHandles were a range of consecutive 0-based integers [0, 1, 2, ...).  
* In ShiftDriver, the CellHandles will be a range of consecutive 0-based integers, since they are the consectutive 0-based global cell handles from SHIFT itself.  For that reason, ShiftDriver can implement `cells_` as a vector and use `cell_.at(handle)` to get the internal cell object for a given CellHandle.  
* In OpenmcDriver, I'm generating the CellHandles from a Cantor pairing of cell index and instance.  The cell indices and instances are both ranges of consecutive 0-based integers, then the Cantor pairing would also be a range of consectutive 0-based of integers.  However, I wasn't sure if I could assume that about the cell indices and instances.  Hence, in OpenmcDriver, I made `cells_` a map rather than a vector.  That way, we can still use `cells_.at(handle)` to get the internal cell object for a given CellHandle.  

Perhaps the different implementations of `cells_` make a bit more sense in the respective functions.  Consider the implementations of `ShiftDriver::set_density` and `OpenmcDriver::set_density`:

```
void ShiftDriver::set_density(CellHandle handle, double rho) const
{
  Expects(rho > 0);
  auto cell = cells_.at(handle);
  Expects(cell >= 0 && cell < this->n_cells());
  int matid = geometry_->matid(cell);
  driver_->compositions()[matid]->set_density(rho);
}

void OpenmcDriver::set_density(CellHandle cell, double rho) const
{
  cells_.at(cell).material()->set_density(rho, "g/cm3");
}
```

In both cases, we're using `cells_.at(i)` to get the neutronics driver's internal cell object that corresponds to a given CellHandle `i`.  (Sorry that that the two functions use different names `handle` and `cell` for the same parameter!)

Reply:
See also:  https://github.com/enrico-dev/enrico/pull/131/files#r607202634

Reply:
Thanks for the detailed explanation! That makes perfect sense now -- we can view both `cells_` members as being mappings from cell handle → internal cell type, only one of which is implemented as an actual `map`.

Reply:
Awesome!  I will change those parameter names to make the correspondence bit clearer, too.

Reply:
`NeutronicsDriver` names the parameters the same way as `OpenmcDriver, so I'm just leaving the param names as-is here.